library/std/src/thread/mod.rs

```diff
@@ -125,9 +125,12 @@
 //! ## Stack size
 //!
 //! The default stack size is platform-dependent and subject to change.
-//! Currently, it is 2 MiB on all Tier-1 platforms.
+//! Currently, it is 2 MiB on all Tier-1 platforms to make programs behave more consistently
+//! cross-platform (notably Windows would otherwise default to 1MiB stacks).
```

Comment:
Hm, maybe this is a nitpick but the Windows comment isn't strictly accurate. The default size is linker-defined, which does (on MSVC at least) use 1MiB if not otherwise set but it can be different. So saying 0 means 1MiB on Windows is not necessarily true even now and in the future Rust might opt to use a different value by default.

That said, I'm not sure it's worth going on about [`/stack`](https://learn.microsoft.com/en-us/cpp/build/reference/stack-stack-allocations?view=msvc-170) here, Maybe it's acceptable to say "1MiB" in a lies to children kind of way.

Reply:
Yeah that's basically the emotional arc I went through. I was sorely tempted to specifically "usually" or something.

Reply:
Fair. I'm a little bit uncomfortable with std docs that aren't *technically correct* but it is truthish and it does make sense to mention the 1MiB somewhere as it's something that does trip up cross-platform code (admittedly mostly for the main thread rather than spawning new ones but still).